--- a/drivers/net/wireless/mediatek/mt76/mt7915/init.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7915/init.c
@@ -405,12 +405,8 @@ mt7915_init_wiphy(struct mt7915_phy *phy
 		phy->mt76->sband_2g.sband.ht_cap.cap |=
 			IEEE80211_HT_CAP_LDPC_CODING |
 			IEEE80211_HT_CAP_MAX_AMSDU;
-		if (is_mt7915(&dev->mt76))
-			phy->mt76->sband_2g.sband.ht_cap.ampdu_density =
-				IEEE80211_HT_MPDU_DENSITY_4;
-		else
-			phy->mt76->sband_2g.sband.ht_cap.ampdu_density =
-				IEEE80211_HT_MPDU_DENSITY_2;
+		phy->mt76->sband_2g.sband.ht_cap.ampdu_density =
+			IEEE80211_HT_MPDU_DENSITY_4;
 	}
 
 	if (phy->mt76->cap.has_5ghz) {
@@ -420,11 +416,10 @@ mt7915_init_wiphy(struct mt7915_phy *phy
 		phy->mt76->sband_5g.sband.ht_cap.cap |=
 			IEEE80211_HT_CAP_LDPC_CODING |
 			IEEE80211_HT_CAP_MAX_AMSDU;
+		phy->mt76->sband_5g.sband.ht_cap.ampdu_density =
+			IEEE80211_HT_MPDU_DENSITY_4;
 
 		if (is_mt7915(&dev->mt76)) {
-			phy->mt76->sband_5g.sband.ht_cap.ampdu_density =
-				IEEE80211_HT_MPDU_DENSITY_4;
-
 			vht_cap->cap |=
 				IEEE80211_VHT_CAP_MAX_MPDU_LENGTH_7991 |
 				IEEE80211_VHT_CAP_MAX_A_MPDU_LENGTH_EXPONENT_MASK;
@@ -434,9 +429,6 @@ mt7915_init_wiphy(struct mt7915_phy *phy
 					IEEE80211_VHT_CAP_SHORT_GI_160 |
 					FIELD_PREP(IEEE80211_VHT_CAP_EXT_NSS_BW_MASK, 1);
 		} else {
-			phy->mt76->sband_5g.sband.ht_cap.ampdu_density =
-				IEEE80211_HT_MPDU_DENSITY_2;
-
 			vht_cap->cap |=
 				IEEE80211_VHT_CAP_MAX_MPDU_LENGTH_11454 |
 				IEEE80211_VHT_CAP_MAX_A_MPDU_LENGTH_EXPONENT_MASK;
