--- a/drivers/net/dsa/mt7530.c
+++ b/drivers/net/dsa/mt7530.c
@@ -1161,6 +1161,9 @@ mt753x_cpu_port_enable(struct dsa_switch
 	 */
 	mt7530_write(priv, MT7530_PCR_P(port),
 		     PCR_MATRIX(dsa_user_ports(priv->ds)));
+		     
+	/* Enable learning on CPU port */
+	mt7530_rmw(priv, MT7530_PSC_P(port), SA_DIS, 0);
 
 	/* Set to fallback mode for independent VLAN learning */
 	mt7530_rmw(priv, MT7530_PCR_P(port), PCR_PORT_VLAN_MASK,
