--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -3370,12 +3370,6 @@ static netdev_features_t mtk_fix_feature
 		}
 	}
 
-	if ((features & NETIF_F_HW_VLAN_CTAG_TX) && netdev_uses_dsa(dev)) {
-		netdev_info(dev, "TX vlan offload cannot be enabled when dsa is attached.\n");
-
-		features &= ~NETIF_F_HW_VLAN_CTAG_TX;
-	}
-
 	return features;
 }
 
