--- a/mt76_connac.h
+++ b/mt76_connac.h
@@ -260,6 +260,12 @@ static inline bool is_connac_v1(struct m
 	return is_mt7615(dev) || is_mt7663(dev) || is_mt7622(dev);
 }
 
+static inline bool is_connac_v2(struct mt76_dev *dev)
+{
+	return is_mt7915(dev) || is_mt7916(dev) ||
+	       is_mt7981(dev) || is_mt7986(dev);
+}
+
 static inline bool is_mt76_fw_txp(struct mt76_dev *dev)
 {
 	switch (mt76_chip(dev)) {
